### PR TITLE
GHA container test: use medium size runner and fix slow image build issue in test

### PR DESCRIPTION
**Pull request by huikang**
### Description
- Increase timeout for waiting for container
- Avoid build on-the-fly image, which runs `apt install` for every consul connect container. These are the same containers repeatedly built in a single test case.
- downsize all container-based tests to medium-sized runner

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

**Comment by huikang** [comment]
@NiniOak , thanks for help review the PR. I will post the link to our internal channel.

File: .github/workflows/test-integrations.yml [hunk 1]
```diff
@@ -405,6 +405,8 @@ jobs:
 
       - name: Build consul:local image
         run: docker build -t ${{ env.CONSUL_LATEST_IMAGE_NAME }}:local -f ./build-support/docker/Consul-Dev.dockerfile .
+      - name: Build consul-envoy:target-version image
+        run: docker build -t consul-envoy:target-version --build-arg CONSUL_IMAGE=${{ env.CONSUL_LATEST_IMAGE_NAME }}:local --build-arg ENVOY_VERSION=1.24.6 -f ./test/integration/consul-container/assets/Dockerfile-consul-envoy ./test/integration/consul-container/assets
```
Comment (NiniOak) [inline .github/workflows/test-integrations.yml:409]: Should we pass the envoy version as a parameter here? `ENVOY_VERSION=1.24.6`

File: .github/workflows/test-integrations.yml [hunk 2]
```diff
@@ -538,6 +540,10 @@ jobs:
         run: chmod +x consul
       - name: Build consul:local image
         run: docker build -t ${{ env.CONSUL_LATEST_IMAGE_NAME }}:local -f ./build-support/docker/Consul-Dev.dockerfile .
+      - name: Build consul-envoy:latest-version image
+        run: docker build -t consul-envoy:latest-version --build-arg CONSUL_IMAGE=docker.mirror.hashicorp.services/${{ env.CONSUL_LATEST_IMAGE_NAME }}:${{ env.CONSUL_LATEST_VERSION }} --build-arg ENVOY_VERSION=1.24.6 -f ./test/integration/consul-container/assets/Dockerfile-consul-envoy ./test/integration/consul-container/assets
```
Comment (NiniOak) [inline .github/workflows/test-integrations.yml:544]: Same idea with the ENVOY_VERSION here

File: .github/workflows/test-integrations.yml (hunk 1 above)
Comment (huikang) [reply .github/workflows/test-integrations.yml]: Done

File: .github/workflows/test-integrations.yml (hunk 2 above)
Comment (huikang) [reply .github/workflows/test-integrations.yml]: Done

**Review by NiniOak — APPROVED** [review]

**Pull request closed without merging**